angularSTEP/src/app/signup/signup.component.ts

```diff
@@ -0,0 +1,123 @@
+import {Component, OnInit} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+
+import {AngularFireAuth} from '@angular/fire/auth';
+import {AngularFirestore} from '@angular/fire/firestore';
+import {Router} from '@angular/router';
+
+interface User {
+  username: string;
+  displayName: string;
+  email: string;
+  photo: string;
+  following: Array<string>;
+  recipes: Array<string>;
+  wishlist: Array<string>;
+  shoppinglist: Array<string>;
+}
+
+@Component({
+  selector: 'app-signup',
+  templateUrl: './signup.component.html',
+  styleUrls: ['./signup.component.scss'],
+})
+export class SignupComponent implements OnInit {
```

Comment:
What is the difference between this and the setup component? They look very similar.